src/components/HeaderSearch/index.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState, useEffect, createRef } from 'react'
+import PropTypes from 'prop-types'
+import { Input, Icon, AutoComplete } from 'antd'
+import classNames from 'classnames'
+import Debounce from 'lodash-decorators/debounce'
+import Bind from 'lodash-decorators/bind'
+import styles from './index.less'
+
+// TODO: define on next task
+// static propTypes = {
+//   className: PropTypes.string,
+//   placeholder: PropTypes.string,
+//   onSearch: PropTypes.func,
+//   onPressEnter: PropTypes.func,
+//   defaultActiveFirstOption: PropTypes.bool,
+//   dataSource: PropTypes.array,
+//   defaultOpen: PropTypes.bool,
+//   onVisibleChange: PropTypes.func
+// }
+
+// static defaultProps = {
+//   defaultActiveFirstOption: false,
+//   onPressEnter: () => {},
+//   onSearch: () => {},
+//   className: '',
+//   placeholder: '',
+//   dataSource: [],
+//   defaultOpen: false,
+//   onVisibleChange: () => {}
+// }
+
+export default function HeaderSearch (props) {
+  const [searchMode, setSearchMode] = useState(props.defaultOpen)
+  const [value, setValue] = useState('')
+  // TO Verify: onVisibleChange not defined or passed as props
+  const { className, placeholder, open, onChange, onPressEnter, ...restProps } = props
+  let textInput = React.createRef()
+
+  let timeout
+
+  useEffect(() => {
+    if (searchMode) {
+      if (textInput.current) {
+        textInput.current.focus()
+      }
+    }
+    // Specify how to clean up after this effect:
+    return function cleanup () {
+      // componentdidunmount
```

Comment:
let's remove this comment, and the comment above. If you want you can add only one, eg `function to cleanup the component when unmounting` or something like that.

Reply:
It is explanatory with the function names. Removed extra comments here and other references in repo.

Reply:
:+1: